### PR TITLE
[Indexing options] Changes for find collections to return indexing option

### LINKED ISSUE #769: [Indexing options] Add indexing options to findCollection response
When findCollections command is run with option {"explain" : true } return indexing option if specified.

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Changes for find collections to return indexing option

**Which issue(s) this PR fixes**:
Fixes #769 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java [hunk 1]
```diff
@@ -34,7 +34,8 @@ public record CreateCollectionCommand(
   public record Options(
 
       // limit of returned documents
-      @Schema(
+      @JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java:37]: We could also use `JsonInclude.Include.NON_EMPTY` which would exclude empty `Collection`s.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java [hunk 2]
```diff
@@ -109,14 +112,35 @@ public CommandResult get() {
...
                             new CreateCollectionCommand.Options.VectorSearchConfig(
                                 collectionProperty.vectorSize(),
                                 collectionProperty.similarityFunction().name().toLowerCase());
+                      }
+
+                      if (collectionProperty.indexingConfig() != null) {
+                        indexingConfig =
+                            new CreateCollectionCommand.Options.IndexingConfig(
+                                collectionProperty.indexingConfig().allowed() != null
+                                        && !collectionProperty.indexingConfig().allowed().isEmpty()
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java:126]: If using `NON_EMPTY` inclusion earlier, could avoid checks here I think? (so `null` and empty Collection would just not be serialized.

**Review by tatu-at-datastax — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java (hunk 2 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java:126]: The IndexingConfig object in the cache has the allow and deny set initialized to empty, I did it so we don't have to do null check when trying to do contains check. Here I need to do not empty check so indexing options don't show up in findCollections as below
```
"indexing": {
  "allow" : [],
  "deny" : []
}
```

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java (hunk 1 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java:37]: Ah., got it I will change it.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindCollectionsIntegrationTest.java [hunk 3]
```diff
@@ -304,6 +309,77 @@ public void notExistingNamespace() {
...
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.collections", hasSize(4))
+          .body(
+              "status.collections",
+              containsInAnyOrder(
```
Comment (Yuqi-Du) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindCollectionsIntegrationTest.java:377]: I thought we only expect 'expected4' here.
No need for expected1, expected2, expected3.

Comment (maheshrajamani) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindCollectionsIntegrationTest.java:377]: find collection returns all the collection created thus far. So add all 4.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java (hunk 2 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java]: Ok yes I realized it after reading later code. Yes, it simplifies later check so everything is fine as-is.

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1: Nice to know about the findCollections command

**Pull request merged**